scripts/linters/pre_commit_linter.py

```diff
@@ -0,0 +1,517 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pre-commit script for Oppia.
+
+This script lints Python and JavaScript code, and prints a
+list of lint errors to the terminal. If the directory path is passed,
+it will lint all Python and JavaScript files in that directory; otherwise,
+it will only lint files that have been touched in this commit.
+
+This script ignores all filepaths contained within .eslintignore.
+
+=====================
+CUSTOMIZATION OPTIONS
+=====================
+1.  To lint only files that have been touched in this commit
+        python -m scripts.linters.pre_commit_linter
+
+2.  To lint all files in the folder or to lint just a specific file
+        python -m scripts.linters.pre_commit_linter --path filepath
+
+3.  To lint a specific list of files (*.js/*.py only). Separate files by spaces
+        python -m scripts.linters.pre_commit_linter
+            --files file_1 file_2 ... file_n
+
+4.  To lint files in verbose mode
+        python -m scripts.linters.pre_commit_linter --verbose
+
+5. To lint a specific list of file extensions (*.js/*.py only). Separate file
```

Comment:
I think you're supporting more extensions than this now, please update the comment (or remove it altogether if the linter now covers all possible files).

Also could you please check whether the specification in point 3 above is up to date, and modify it accordingly?

Finally, can we do one of these formats instead:

-  `--only-check-file-extensions py js`
- `--only-check-file-extensions *.py *.js` (for this, add validation to ensure that args start with *. and provide a clear error message if not).

I think that helps a bit with clarity -- the first option is the actual file extension and the second option gives more clarity about what this is checking if the developer is familiar with the use of *.

Reply:
Done.